galasa-ui/src/components/test-runs/TestRunDetails.tsx

```diff
@@ -89,7 +89,7 @@ const TestRunDetails = ({ runId, runDetailsPromise, runLogPromise, runArtifactsP
     };
 
     loadRunDetails();
-  }, [runDetailsPromise, runArtifactsPromise, runLogPromise]);
+  }, [runDetailsPromise, runArtifactsPromise, runLogPromise, extractRunDetails]);
```

Comment:
Instead of adding more dependencies here, we could just define extractRunDetails inside the useEffect? Assuming it won't be used outside